argmin/src/core/executor.rs

```diff
@@ -26,7 +26,7 @@ pub struct Executor<O, S, I> {
     /// Storage for observers
     observers: Observers<I>,
     /// Checkpoint
-    checkpoint: Option<Box<dyn Checkpoint<S, I>>>,
+    checkpoint: Option<Box<dyn Checkpoint<S, I> + Send>>,
```

Comment:
I added these bounds because `PyClass` must be `Send`.

Reply:
Sounds also good to me. Would it make sense to add the `Send` bound to the `Checkpoint` trait?